### PR TITLE
Generating sorting subroutines specific to `character` type with fypp

### PR DESCRIPTION
**Pull request by jvdp1**
Previously all sorting subroutines for `integer`, `real`, and `string` were generated with `fypp`. `character` had its own subroutines.

I noticed that some checks were not present in `character`-based subroutines, while present for other types. Therefore, I modified the fypp instructions, such that the sorting subrotuines are generated from a same template for `integer`, `real`, `string`, and `character`.

With this strategy, if someone implements a derived type associated with `>`, `<`, `==`,...., the sorting subroutines can be easily implemented for this DT by modifying only the fypp instructions.

Additional reviewer: @wclodius2

### CODE REVIEW

**Review by awvwgk — APPROVED** [review]
I'm always happy to negative line count patch. Looks good to me.

File: src/stdlib_sorting.fypp
Comment (gareth-nx) [inline src/stdlib_sorting.fypp:2]: Suggest you add a comment explaining these fypp definitions. Something along the lines of `For better code reuse in fypp, make lists that contain the input types, with each having output types and a separate name prefix for subroutines`.

Comment (gareth-nx) [reply src/stdlib_sorting.fypp:2]: Also another sentence about why this makes sense would be good (noting it greatly reduces the code, yet is not entirely obvious).

File: src/stdlib_sorting_ord_sort.fypp
Comment (gareth-nx) [inline src/stdlib_sorting_ord_sort.fypp:2]: As above, please add a comment explaining the fypp definitions.

File: src/stdlib_sorting_ord_sort.fypp
Comment (gareth-nx) [inline src/stdlib_sorting_ord_sort.fypp:127]: Here it might be better to check more characters (e.g.  "char" instead of "c") from a clarity perspective.

File: src/stdlib_sorting_sort.fypp
Comment (gareth-nx) [inline src/stdlib_sorting_sort.fypp:2]: As noted earlier, please comment here.

File: src/stdlib_sorting_sort_index.fypp
Comment (gareth-nx) [inline src/stdlib_sorting_sort_index.fypp:2]: Please add comment explaining these

File: src/stdlib_sorting_sort_index.fypp
Comment (gareth-nx) [inline src/stdlib_sorting_sort_index.fypp:128]: As noted previously -- consider comparing with "char" instead of "c"

**Review by gareth-nx — APPROVED** [review]
This is really nice -- in addition to the code simplification I think there are lessons on using fypp that would be nice to share more broadly.

File: src/stdlib_sorting_ord_sort.fypp
Comment (jvdp1) [reply src/stdlib_sorting_ord_sort.fypp:127]: True. One additional advantage is that more considered letters will result in less potential conflicts in the futute (e.g., in the case that we want add a new DT that starts with "c")

File: src/stdlib_sorting.fypp
Comment (jvdp1) [reply src/stdlib_sorting.fypp:2]: Done

File: src/stdlib_sorting_ord_sort.fypp
Comment (jvdp1) [reply src/stdlib_sorting_ord_sort.fypp]: Done

File: src/stdlib_sorting_sort_index.fypp
Comment (jvdp1) [reply src/stdlib_sorting_sort_index.fypp:2]: Done

File: src/stdlib_sorting_sort_index.fypp
Comment (jvdp1) [reply src/stdlib_sorting_sort_index.fypp]: Done

**Comment by jvdp1** [comment]
@gareth-nx Thank you for your review. I believed I answered all your comments.

**Comment by gareth-nx** [comment]
@jvdp1 Thanks for the changes.

If anyone with commit rights has time to assist, note this patch seems very straightforward (it is only code simplification) and has been reviewed twice -- but we need another reviewer with commit rights to move it forward. @milancurcic @ivan-pi

**Comment by jvdp1** [comment]
@milancurcic @ivan-pi @LKedward Could one of you review this PR, and merge it if you agree with it, please?

**Comment by awvwgk** [comment]
With two approvals this patch is ready to go. I'll go ahead and merge it now.

**Pull request merged**
